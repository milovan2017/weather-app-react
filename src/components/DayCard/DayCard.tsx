import React from "react";
import { DayData } from "../../types";
import { Col } from "react-bootstrap";

interface IDayCardProps {
  dayInfo: DayData;
}

export const DayCard: React.FC<IDayCardProps> = ({ dayInfo }) => {
  return (
    <Col className="p-0 text-center">
      <span className="text-muted">{dayInfo.day}</span>
      <h2 className="text-light">{dayInfo.averageTemp}Â°</h2>
    </Col>
  );
};
