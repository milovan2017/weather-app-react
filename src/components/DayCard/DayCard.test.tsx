import React from "react";
import { DayCard } from "./DayCard";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup);

const mockDayData = {
  day: "Wednesday",
  averageTemp: 12,
};

describe("DayCard.tsx", () => {
  it("renders DayCard without crashing", () => {
    const { container } = render(<DayCard dayInfo={mockDayData} />);
    expect(container).toMatchSnapshot();
  });

  it("contains day info", () => {
    const { container } = render(<DayCard dayInfo={mockDayData} />);
    expect(container.textContent).toContain(mockDayData.day.toString());
  });

  it("contains average temperature info", () => {
    const { container } = render(<DayCard dayInfo={mockDayData} />);
    expect(container.textContent).toContain(mockDayData.averageTemp.toString());
  });
});
