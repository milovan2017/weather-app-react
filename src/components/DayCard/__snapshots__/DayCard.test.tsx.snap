// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayCard.tsx renders DayCard without crashing 1`] = `
<div>
  <div
    class="p-0 text-center col"
  >
    <span
      class="text-muted"
    >
      Wednesday
    </span>
    <h2
      class="text-light"
    >
      12
      Â°
    </h2>
  </div>
</div>
`;
